### PR TITLE
options/posix: fix getaddrinfo() service resolution

### PR DESCRIPTION
**Pull request by Geertiebear**

### CODE REVIEW

File: tests/posix/getaddrinfo.c [hunk 1]
```diff
@@ -21,5 +21,12 @@ int main() {
 	assert(res[0].ai_protocol == IPPROTO_TCP);
 
 	freeaddrinfo(res);
+
+	/* check we can resolve any domain */
+	ret = getaddrinfo("example.net", NULL, &hints, &res);
+	assert(ret == 0);
+
+	freeaddrinfo(res);
```
Comment (ArsenArsen) [inline tests/posix/getaddrinfo.c:29]: there is a double-free here

File: options/posix/generic/services.cpp [hunk 2]
```diff
@@ -83,39 +84,43 @@ static int lookup_serv_file(struct service_buf *buf, const char *name,
...
 int lookup_serv(struct service_buf *buf, const char *name, int proto,
-		int socktype) {
+		int socktype, int flags) {
 	switch(socktype) {
 		case SOCK_STREAM:
 			if (!proto)
 				proto = IPPROTO_TCP;
 			else if (proto != IPPROTO_TCP)
-				return EAI_SERVICE;
+				return -EAI_SERVICE;
```
Comment (avdgrinten) [inline options/posix/generic/services.cpp:93]: Why the minus?

Comment (Geertiebear) [reply options/posix/generic/services.cpp:93]: A positive return value indicates success, so I return a negative one, and then make it positive again in ``getaddrinfo()``

Comment (avdgrinten) [reply options/posix/generic/services.cpp:93]: Okay, maybe add a comment to state that it works like this (?).

Other than that, the PR LGTM.

**Pull request merged**